README.md

```diff
@@ -47,6 +48,102 @@ programming resources.
 * [Flawed Gems](#flawed-gems)
 * [Managing processes](#managing-processes)
 
+## Testing
+### Rspec Unit Tests
+* <a name="unit-describe"></a>
+  There should be a `describe` block for each method. For class methods, the `describe` statement should 
+  be `.method_name`, and for instance methods `#method_name`.
+<sup>[[link](#unit-describe)]</sup>
+
+* <a name="describe-valid-scopes"></a>
+  Use a new `describe` block for any validations or scopes that are tested.
+<sup>[[link](#describe-valid-scopes)]</sup>
+
+* <a name="don't-test-privates"></a>
+  Don't test private methods.
+<sup>[[link](#don't-test-privates)]</sup>
+
+* <a name="use-context"></a>
+  Complex or multiple `it` statements should go in a `context` block (a single, short `it` can stand alone).
+<sup>[[link](#use-context)]</sup>
+
+* <a name="context-when-with-given"></a>
+  Context blocks should almost always begin with one of three words "when", "with" or "given".
+<sup>[[link](#context-when-with-given)]</sup>
+
+  ````
+  context "when the user is not logged in"
+  context "with an assignment record that is overdue"
+  context "given an overdue assignment"
+  ````
+
+* <a name="don't-test-code"></a>
+  For both `context` statements and `it` statements, make sure you can easily identify the purpose of the 
+  spec, and that it describes the behavior desired, and not simply the code being tested.
+<sup>[[link](#don't-test-code)]</sup>
+
+  Instead of:
+  ````
+  context "when param next_transfer_date is >= Date.today"
+    it "returns current_section_attempts + old_section.attempts"
+  ````
+  Do something like:
+  ````
+  context "when the user will be transfered today or later"
+    it "returns the combined attempts from their current section and their old section"
+  ````
+
```

Comment:
New item:

Don't do data setup in a before block when a context has only one "it" block.  Move the data setup to inside the "it" block.